src/core/futures.rs

```diff
@@ -23,7 +23,10 @@ macro_rules! fry {
     ($res:expr) => {
         match $res {
             Ok(elt) => elt,
-            Err(e) => return futures::done(Err(From::from(e))),
+            Err(e) => {
+                use $crate::core::futures::FutureExt;
+                return futures::done(Err(From::from(e))).into_box()
```

Comment:
Not very sure of this change though - it might help in function return types but will cause extra allocation in combinators:
```
fut.and_then(move |result| {
    let val = fry!(result);
    let final_val = fry!(process(val));
    ok!(final_val);
});
```
So those cases can no longer be written without heap allocation anymore. Where as if `fry!` is left as is, then you always have the option of `into_box()` that you coded to covert it to box.